presto-main/src/main/java/com/facebook/presto/util/HashCollisionsEstimator.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Math.ceil;
+import static java.lang.Math.floor;
+
+public final class HashCollisionsEstimator
```

Comment:
I see. Could you add some comments to this class that describe why this linearization works, and point to the class used to generate the constants? It looks like they were only generated for up to 10k values, and it's not clear to be that we can extrapolate linearly out to say 10 trillion values